src/meta/src/hummock/compactor_manager.rs

```diff
@@ -133,69 +108,57 @@ impl Compactor {
 ///   It's the final state.
 /// - 4. Cancelled: a task is reported as cancelled via `CompactStatus::report_compact_task`. It's
 ///   the final state.
-/// We omit Assigned state because there's nothing to be done about this state currently.
-///
-/// Furthermore, the compactor for a compaction task must be picked with `CompactorManager`,
-/// or its internal states might not be correctly maintained.
-pub struct CompactorManager {
-    policy: RwLock<Box<dyn CompactionSchedulePolicy>>,
-
+pub struct CompactorManagerInner {
     pub task_expiry_seconds: u64,
-    // A map: { context_id -> { task_id -> heartbeat } }
-    task_heartbeats:
-        RwLock<HashMap<HummockContextId, HashMap<HummockCompactionTaskId, TaskHeartbeat>>>,
+    task_heartbeats: HashMap<HummockCompactionTaskId, TaskHeartbeat>,
+
+    /// The context ids of compactors.
+    pub compactors: Vec<HummockContextId>,
```

Comment:
As there is an exact 1-1 mapping between `compactors` and `compactor_map`,  we can just remove `compactors`.

Reply:
yes, its an 1-1 mapping, compactors only used to choose a random_index for `next_compactor` function, any suggestion?